### PR TITLE
Fix squadmanager adding naval units to ground attack forces.

### PR DESCRIPTION
**Pull request by blackhand1001**
Fix squadmanager adding naval units to ground attack forces. This was breaking the behavior of both naval and ground squads.

### CODE REVIEW

File: OpenRA.Mods.Common/Traits/BotModules/SquadManagerBotModule.cs [hunk 1]
```diff
@@ -279,7 +279,7 @@ void TryToRushAttack(IBot bot)
 			// TODO: This should use common names & ExcludeFromSquads instead of hardcoding TraitInfo checks
 			var ownUnits = activeUnits
 				.Where(unit => unit.IsIdle && unit.Info.HasTraitInfo<AttackBaseInfo>()
-					&& !unit.Info.HasTraitInfo<AircraftInfo>() && !unit.Info.HasTraitInfo<HarvesterInfo>()).ToList();
+                    && !unit.Info.HasTraitInfo<AircraftInfo>() && !Info.NavalUnitsTypes.Contains(unit.Info.Name) && !unit.Info.HasTraitInfo<HarvesterInfo>()).ToList();
```
Comment (reaperrr) [inline OpenRA.Mods.Common/Traits/BotModules/SquadManagerBotModule.cs]: Please keep this as tabs instead of spaces.

Comment (blackhand1001) [reply OpenRA.Mods.Common/Traits/BotModules/SquadManagerBotModule.cs]: oh weird. I always use tabs. Not sure why it converted.

Comment (blackhand1001) [reply OpenRA.Mods.Common/Traits/BotModules/SquadManagerBotModule.cs]: Strange its showing as tabs on my local file.

Comment (blackhand1001) [reply OpenRA.Mods.Common/Traits/BotModules/SquadManagerBotModule.cs]: Ok fixed. Also added in one more spot to check for naval.

**Comment by GraionDilach** [comment]
This limits amphibious units to be used only for ground or naval at AI and prevents using them for both though.

**Comment by blackhand1001** [comment]
Without this fix naval squads are completely broken. They end up being controlled by two squads and the two squads fight back and forth to give them orders. Also the units in naval units are not hardcoded. You can set which ones are considered naval units.

**Comment by GraionDilach** [comment]
If I add navy to TS, I expect the Hover MRLSes I have to be used for both, to give the AI more chance of fighting back in the water while not losing the most important AA on the ground.

You're substituting one bug for another and you can easily just add an `AmphibiousUnitsList` which is included in both if you're already touching this.

Also, if I read the PR correctly, you also break shore attacks as well, but I don't have an easy fix for taht one.

**Comment by blackhand1001** [comment]
Amphibious units being added to a naval squad would end up just going across the map normally and driving through the enemy base on any map that has the water around the border.

**Comment by GraionDilach** [comment]
Whatever then, I just work around downstream as I tend to.

> Also the idleunitscan will still target enemy boats that's the attacking the base as long as it can reach them so the hovers should have no issue attacking the enemy ships.

This worked so well for medics/mechanics already. #11981

**Review by ghost — APPROVED** [review]

**Review by abcdefg30 — APPROVED** [review]
Looks like this really does improve naval play of the AI a lot.

**Comment by abcdefg30** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/b133660a55360a6d85ed7f1be9b1f651f2916423...343315314ec6782b9891da71768f8fe59c5afd7f)

**Comment by pchote** [comment]
This was picked to prep as 93e408364018b643efc615dfbe0a5636387d2639, changelog https://github.com/OpenRA/OpenRA/wiki/Changelog/_compare/2c76f8e60815891c77e2aedbbfa1a111d40a8dd6

**Pull request merged**